application/src/objects/obstacles/abstract-dynamic-obstacle.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+var Phaser = require('phaser');
+
+var AbstractDynamicObstacle = function(state, x, y, key, collisionGroup)
+{
+    Phaser.Sprite.apply(this, [state.game, x, y, key]);
+
+    this.createPhysicsBody(state);
+
+    this.body.setCollisionGroup(collisionGroup);
+    this.body.mass = this.getAttributes().mass;
+    this.body.angularDamping = this.getAttributes().angularDamping;
+};
+
+AbstractDynamicObstacle.prototype = Object.create(Phaser.Sprite.prototype);
+
+AbstractDynamicObstacle.prototype.getSpritePath = function()
+{
+    throw new Error('Attempted to load assets on abstract class');
+};
+
+AbstractDynamicObstacle.prototype.getAttributes = function()
```

Comment:
Maybe I'm getting to far ahead of things, but I was reluctant to call these attributes 'constants', since I could see a situation in which they could be modified (Driving over a patch of ice? A power up/power down that changes your car's mass?), but maybe it does make sense to preserve these as 'base values' and make them constants.

As for CONSTAN_CASE, I can go along with that, for sure. How would we do that with the function approach? Have the object keys be all caps? The GET_CONSTANTS function name?